### PR TITLE
Feature/support code analysis

### PR DESCRIPTION
**Pull request by ElijahGeorge**
If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### What does this PR do?
It provides support for generating static analysis and complexity metrics for the code base.

#### Related JIRA tickets:
https://jira.amida-tech.com/browse/DEVOPS-168

#### How should this be manually tested?
See `Code Analysis` section in `README.md`

### CODE REVIEW

**Comment by unprofessional** [comment]
@jsachs Jacob, I'm giving you this as the primary reviewer since it concerns a DevOps analysis integration. Assigning @au2 Afsin as secondary for awareness of this change.

**Comment by au2** [comment]
1) Artifacts directory should go to .gitignore.
2) Lint options should not contradict .eslintrc.js in this project. I see indentation 2 for example. Better yet lint options should be read from .eslintrc.js.

**Comment by ElijahGeorge** [comment]
Good catch @au2. Will be pushing these changes shortly.

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    'sourceType': 'module',
+    'ecmaFeatures': {
+      'jsx': true,
+      'modules': true
+    }
+  }
+};
+
+
+let complexityRules = {
```
Comment (jsachs) [inline gulpfile.js]: Do we need the empty object?

**Comment by unprofessional** [comment]
Conflicts need to be resolved before this can be merged.

**Comment by jsachs** [comment]
@ElijahGeorge please resolve

**Pull request merged**